extensions/oplogPopulator/OplogPopulator.js

```diff
@@ -77,30 +127,262 @@ class OplogPopulator {
+     * @throws {InternalError}
+     */
+    async _getBackbeatEnabledBuckets() {
+        const filter = {
+            $or: [],
+        };
+        // setup filters
+        this._activeExtensions.forEach(extName => {
+            const field = constants.extensionNameToBucketMetadataConfigField[extName];
+            if (!field) {
+                return;
+            }
+            const extFilter = {};
+            // extension configuration should be an object (BSON type 3) not null
+            extFilter[`value.${field}`] = { $type: 3 };
+            filter.$or.push(extFilter);
+        });
+        try {
+            const buckets = await this._metastore.find(filter)
+                .project({ _id: 1 })
+                .map(bucket => bucket._id)
+                .toArray();
+            return buckets;
+        } catch (err) {
+            this._logger.error('Could not connect to MongoDB', {
+                method: 'OplogPopulator._setupMongoClient',
+                error: err.message,
+            });
+            throw errors.InternalError.customizeDescription(err.message);
+        }
+    }
+
+    /**
+     * Makes new connector pipeline to listen to
+     * specified buckets
+     * @param {string[]} bucketNames list of buckets to listen to
+     * @returns {Object} new connector pipeline
+     */
+    _generateNewConnectorPipeline(bucketNames) {
+        const pipeline = [
+            {
+                $match: {
+                    'ns.coll': {
+                        $in: bucketNames,
+                    }
+                }
+            }
+        ];
+        return JSON.stringify(pipeline);
+    }
+
+    /**
+     * Check if buckets has at least one backbeat extension active
+     * @param {BucketInfo} bucketMetadata bucket metadata
+     * @returns {boolean} is bucket backbeat enabled
+     */
+    _isBucketBackbeatEnabled(bucketMetadata) {
+        const areExtensionsEnabled = [];
+        this._activeExtensions.forEach(extName => {
+            const field = constants.extensionNameToBucketMetadataConfigField[extName];
+            if (!field) {
+                return;
+            }
+            areExtensionsEnabled.push(bucketMetadata[field] !== null);
+        });
+        return areExtensionsEnabled.some(ext => ext);
+    }
+
+    /**
+     * Handler for the change stream "change" event.
+     * Updated connector pipeline when change occurs
+     * @param {ChangeStreamDocument} change Change stream change object
+     * @returns {undefined}
+     */
+    _handleChangeStreamChangeEvent(change) {
+        // caching change stream resume token
+        this._changeStreamResumeToken = change._id;
+        // updating list of buckets to listen to
+        const idx = this._bucketsToListenTo.indexOf(change.documentKey._id);
+        // no fullDocument field in delete events
+        const isBackbeatEnabled = change.fullDocument ? this._isBucketBackbeatEnabled(change.fullDocument.value)
```

Comment:
This will cause an issue if a bucket becomes invalid (no extension active), as the change stream event that invalidates it will be ignored and so the connector pipeline won't be updated